src/Microsoft.AspNetCore.Server.Kestrel/Http/DateHeaderValueManager.cs

```diff
@@ -65,7 +64,7 @@ public DateHeaderValues GetDateHeaderValues()
 
             return new DateHeaderValues()
             {
-                Bytes = _activeDateBytes ? _dateBytes0 : _dateBytes1,
+                Bytes = _dateBytes,
```

Comment:
> It seems like it's possible (though very unlikely) that you could get mismatching values for the bytes and the string, no?

They could differ by 1 sec every 1 sec for a window of ~100ns. Only the byte representation is actually used; except if someone creates a middleware to read it internally for some reason. 

However for timings and diagnostics you would much more likely use the more accurate [timestamps](https://github.com/aspnet/Hosting/blob/b7bdc9c40494f8e0f0eac22db91b8d5c58811ee2/src/Microsoft.AspNetCore.Hosting/Internal/HostingApplication.cs#L37-L44) which give sub millisecond accuracy rather than 1sec +/- 1sec accuracy.

Use of the date header for caching/proxy/age normally is for time periods of between 5 min and 1 year so the accuracy isn't too huge a factor.

Changing it to a class would add an extra level of indirection per request; and an extra allocation (though only once per second!).

Happy to change it if there is a strong opinion on it.